dags/glam_fenix.py

```diff
@@ -67,153 +71,164 @@
 
 tags = [Tag.ImpactTier.tier_1]
 
-dag = DAG(
+with DAG(
     "glam_fenix",
     default_args=default_args,
     max_active_runs=1,
     schedule_interval="0 2 * * *",
     doc_md=__doc__,
     tags=tags,
-)
-
-wait_for_copy_deduplicate = ExternalTaskCompletedSensor(
-    task_id="wait_for_copy_deduplicate",
-    external_dag_id="copy_deduplicate",
-    external_task_id="copy_deduplicate_all",
-    execution_delta=timedelta(hours=1),
-    check_existence=True,
-    mode="reschedule",
-    pool="DATA_ENG_EXTERNALTASKSENSOR",
-    email_on_retry=False,
-    dag=dag,
-)
-
-pre_import = DummyOperator(
-    task_id='pre_import',
-    dag=dag,
-)
-
-mapping = {}
-for product in PRODUCTS:
-    query = generate_and_run_glean_queries(
-        task_id=f"daily_{product}",
-        product=product,
-        destination_project_id=PROJECT,
-        env_vars=dict(STAGE="daily"),
+) as dag:
+    wait_for_copy_deduplicate = ExternalTaskSensor(
+        task_id="wait_for_copy_deduplicate",
+        external_dag_id="copy_deduplicate",
+        external_task_id="copy_deduplicate_all",
+        execution_delta=timedelta(hours=1),
+        check_existence=True,
+        mode="reschedule",
+        allowed_states=ALLOWED_STATES,
+        failed_states=FAILED_STATES,
+        pool="DATA_ENG_EXTERNALTASKSENSOR",
+        email_on_retry=False,
         dag=dag,
```

Comment:
Do we still need to pass in the DAG via the `dag` parameters everywhere?

I thought one reason for using the context manager approach is that the DAG then automatically gets applied to everything in that context scope and doesn't need to be passed in.